### DISCUSSION TITLE
Macros in API rules not working as expected

### DISCUSSION THREAD

**Discussion by ivanpajon** [pocketbase/pocketbase#5078] [Q&A] [answered]
I'm trying to define a custom rule that compares sent date in `@request.data` with `@day` macro, as shown in [documentation](https://pocketbase.io/docs/api-rules-and-filters/) (search for `@request.data.publicDate >= @now`), but internally Pocketbase is doing `SELECT DISTINCT ....... AND '2024-06-12 12:00:00' > 14)) LIMIT 1` and never throws an error.
I can't find any other example of this and I don't know if exist something like `@request.data.publicDate.day >= @now`.

**Comment by ganigeorgiev** [comment]
It works as intended (_there are also [automated tests](https://github.com/pocketbase/pocketbase/blob/master/tools/search/identifier_macros_test.go) that verify the expected behavior_).

As per the [docs](https://pocketbase.io/docs/api-rules-and-filters/#-macros), `@now` should return the current datetime as string. `@day` returns only the `@now` day number (eg. 14).


>  I don't know if exist something like `@request.data.publicDate.day >= @now`

I don't understand what the above is supposed to do.

**Reply by ganigeorgiev** [reply]
If you want to check whether a submitted date is "today" (no matter of the time) then you can specify the range:

```js
(@request.data.publicDate >= @todayStart && @request.data.publicDate <= @todayEnd)
```

**Reply by ivanpajon** [reply]
I want to check if a submitted date is greater than today, and I think your response accomplish that. Or if possible, if submitted date is greater than tomorrow or some other day in the future (maybe doing @request.data.publicDate >= (@todayStart + 2) ?).

**Reply by ganigeorgiev** [reply, marked as answer]
No, math expressions currently are not supported.

For "tomorrow" you can check against `@todayEnd`:

```js
@request.data.publicDate > @todayEnd
```

**Reply by ivanpajon** [reply]
Ohhh I see... it would be a nice feature for the future 🙂
Thank you very much for your fast response and the amazing work with Pocketbase!

**Reply by ivanpajon** [reply]
Doing some tests I notice that if `@request.data.publicDate = 2024-06-25T00:00:00.000Z` (which is an ISO 8601 date) and `@todayEnd = 2024-06-25 23:59:59.999Z`, internally in SQL is doing `'2024-06-25T00:00:00.000Z' > '2024-06-25 23:59:59.999Z'`, and this comparison returns true because of 'T' letter in @todayEnd value. So it is accepting a date it shouldn't.

Maybe Pocketbase should follow [ISO 8601](https://es.wikipedia.org/wiki/ISO_8601) standard for dates and add that 'T' letter?

**Reply by ganigeorgiev** [reply]
@ivanpajon The date format that is returned and stored in the database is per rfc3339, aka. `Y-m-d H:i:s.uZ` (it is mostly for backward comparability with earlier versions). Support for custom date format exists in the roadmap but remains a low priority.
